fastlane/docs/Plugins.md

```diff
@@ -0,0 +1,149 @@
+
+- Generally usable for a big majority of developers (e.g. basic interactions with git)
+- Solve pain points for mobile app developers (iOS and Android)
+- have an easy to read documentation and great test coverage
+
+In general, we might not accept actions that
+
+- Solve specific use-cases for only a small subset of developers
+- Access the API of a third party service, the third party service should own and maintain the action
+- Complex actions, that will require a lot of work to maintain in the future
+- Everything that isn't mobile developer related
+
+## Remote plugins
+
+`fastlane` is an open platform and we enable every developer to extend it to fit their needs. That's why we built a plugin system that allows you and your company to provide `fastlane` plugins to other `fastlane` users. You have the full power and responsibility of maintaining your plugin and keeping it up to date. This is useful if you maintain your own library or web service, and want to make sure the `fastlane` plugin is always up to date.
+
+### Find a plugin
+
+Head over the [AvailablePlugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/AvailablePlugins.md) for a list of plugins you can use.
+
+List all available plugins using
+
+```
+fastlane search_plugins
+```
+
+To search for something specific 
+```
+fastlane search_plugins [query]
+```
+
+### Add a plugin to your project
+
+```
+fastlane add_plugin [name]
+```
+
+`fastlane` will assist you on setting up your project to start using plugins.
+
+This will:
+
+- Add the plugin to `fastlane/Pluginfile`
+- Make sure your `fastlane/Pluginfile` is properly referenced from your `./Gemfile`
+- Run `bundle install` to make sure all required dependencies are installed on your local machine (this step might ask for your admin password to install Ruby gems)
+
+#### Plugin Source
+
+Your `fastlane/Pluginfile` contains the list of all `fastlane` plugins your project uses. The `Pluginfile` is a [Gemfile](http://bundler.io/gemfile.html) that gets imported from your main `Gemfile`.
+You specificy all dependencies, including the required version numbers:
+
+```ruby
+# Fetched from RubyGems.org
+gem "fastlane-plugin-xcversion"
+
+# Fetched from GitHub
+gem "fastlane-plugin-xcversion", git: "https://github.com/fastlane/fastlane-plugin-xcversion"
+
+# Fetched from a local directory
+gem "fastlane-plugin-xcversion", path: "../fastlane-plugin-xcversion"
+
+# Specify a version requirements
+gem "fastlane-plugin-xcversion", "1.1.0"
+gem "fastlane-plugin-xcversion", ">= 1.0"
+```
+
+[More information about a Gemfile](http://bundler.io/gemfile.html)
+
+### Install plugins on another machine
+
+To make sure all plugins are installed on the local machine, run
+
+```
+fastlane install_plugins
+```
+
+### Remove a plugin
+
+Open your `fastlane/Pluginfile` and remove the line that looks like this
+
+```
+gem "fastlane-plugin-[plugin_name]"
+```
+
+### Create your own plugin
+
+```
+cd ~/new/folder/[plugin_name]
```

Comment:
Because the folder `[plugin_name]/` will be created by the command, let's leave that off here.